specs/Pills.md

```diff
@@ -0,0 +1,182 @@
+# [Pills]()
+
+## Background
+
+PIlls should be used when representing an input, as a way to filter content, or to represent an attribute.
+
+## Prior Art
+
+- [Evergreen](https://evergreen.segment.com/components/badge-and-pill/)
+- [Lightning](https://www.lightningdesignsystem.com/components/pills/)
+- [Material UI](https://material-ui.com/components/chips/#chip)
+- [OpenUI Research](https://github.com/WICG/open-ui/pull/259)
+
+## Sample Code
+
+```jsx
+<Pill>Item 1</Pill>
+<Pill clickable>Item 2</Pill>
+<TogglePill>On</Pill>
+```
+
+## Variants
+
+- Shape: `circular`, `rounded`
+- Size: `smaller`, `small`, `medium`
+- Appearance: `Filled Grey (filled)`, `Filled White (inverted)`, `Outline`
+
+## PROPS
+
+```typescript
+export type PillShape = 'rounded' | 'circular';
+
+export type PillAppearence = 'filled' | 'inverted' | 'outline';
+
+export type SizeValue = 'smaller' | 'small' | 'medium';
+
+export interface PillProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * A Pill can be sized.
+   */
+  size?: SizeValue;
+
+  /**
+   * A Pill can be circular or rounded
+   */
+  shape?: PillShape;
+
+  /**
+   * A Pill can be filled, inverted or outline
+   */
+  appearance?: PillVariant;
+
+  /**
+   * Media slot
+   */
+  media?: ShorthandProps<HTMLElement>;
+
+  /**
+   * Details that will appear as second line
+   */
+  details?: ShorthandProps<HTMLElement>;
+
+  /** A Pill can be dismissible. */
+  dismissible?: boolean;
+
+  /** A Pill can be disabled. */
+  disabled?: boolean;
+
+  /** A Pill can be selected. */
+  selected?: boolean;
+
+  /** A Pill can be clickable */
+  clickable?: boolean;
+
+  /** A callback to be called when Pill is clicked */
+  onClick?: function;
+
+  /**
+   * A button shorthand for the dismiss action slot. To use this slot the pill should be
+   * dismissible.
+   */
+  dismissIcon?: ShorthandProps<HTMLElement>;
+}
+
+export type TogglePillProps = Omit<PillProps, 'clickable'>;
+```
+
+## Structure
+
+- _**Public**_
+
+```jsx
+<Pill />
+
+<Pill clickable />
+
+<TogglePill />
+```
+
+- _**DOM**_
+
+```html
+<span>
+  ...
+</span>
+
+<span role="button">
+  ...
+</span>
+
+<span role="switch" aria-checked>
+  ...
+</span>
+```
+
+## Migration
+
+- _Migration from v8_
+
+There's no components related to `Pill` in V8
+
+- _Migration from v0_
+
+`Dropdown` component should use `Pills` over `DropdownSelectedItem`.
+
+### Pill with dismiss
+
+#### Accessibility
+
+- Tab key to set focus on the Pill
+- Tab key moves focus to next `Pill`
+- Esc key dismiss the `Pill`
+
+#### Keyboard Navigation
+
+| key                | state    | title                                                              |
+| ------------------ | -------- | ------------------------------------------------------------------ |
+| left/right         | OnFocus  | Role= “button”, aria-describedby=”Press enter or delete to remove” |
```

Comment:
also up/down can work. For roles, listbox/option were proposed. The instruction message needs to come from a prop to allow translation

Reply:
on the line 130, tabbing is proposed to navigate among pill, or is that to navigate among pill containers?

Reply:
It should be to navigate among components, although I recall seeing in the design spec telling about navigating between pills

Reply:
The design spec actually says that should navigate between items

Reply:
During review, listbox/options roles were proposed which are connected with arrow key navigation. If this has not been reflected in the design, we need to ask designers to apply this change.